src/devices/Hx711/HX711.cs

```diff
@@ -0,0 +1,259 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Device;
+using System.Device.Gpio;
+using System.Linq;
+using UnitsNet;
+
+namespace Iot.Device.HX711
+{
+    /// <summary>
+    /// HX711 - Weight scale Module
+    /// </summary>
+    /// <remarks>DataSheet: https://html.alldatasheet.com/html-pdf/1132222/AVIA/HX711/109/1/HX711.html</remarks>
+    public sealed class HX711 : IDisposable
+    {
+        private readonly GpioController _gpioController;
+        private readonly bool _shouldDispose;
+
+        private readonly int _pinDout;
+        private readonly int _pinPD_Sck;
+
+        private readonly HX711Options _options;
+        private readonly object _readLock;
+        private readonly HX711Reader _reader;
+
+        private bool _isInitialize;
+        private bool _isCalibrated;
+
+        /// <summary>
+        /// Conversion ratio between Hx711 units and grams
+        /// </summary>
+        public double ReferanceUnit { get; private set; }
+        private readonly List<double> _referenceUnitList = new();
+
+        /// <summary>
+        /// Offset value from 0 at startup
+        /// </summary>
+        private int _offsetFormZero;
+
+        /// <summary>
+        /// Weight set as tare
+        /// </summary>
+        public Mass TareValue => Mass.FromGrams(Math.Round(_tareValue / ReferanceUnit));

```

Comment:
No, it can't be `null`, but it can be 0 indeed. 
It's 0 when the class is freshly instantiated and then assumes the correct value either via `SetCalibration` or via `SetReferenceUnit`, so I've handled that.